### PR TITLE
fix slickgrid keyboard event handler

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes: #23561
With the recent merge, all the SlickGrid keyboard event handlers are broken due to a change in the StandardKeyboardEvent class: https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/browser/keyboardEvent.ts#L145. previously we would cast the event to KeyboardEvent type but it has always been a JQuery Event, it doesn't have the method `getModifierState`.

the fix is to use the original keyboard event to construct the StandardKeyboardEvent.

this is the error that will be thrown whenever we try to handle the keydown event of SlickGrid.
image

### CODE REVIEW

**Review by aasimkhan30 — APPROVED** [review]
Looks good. Should we have a general helper method that converts any keyboard (both jquery and original) to standard keyboard event.

File: src/sql/base/browser/dom.ts [hunk 1]
```diff
@@ -118,3 +118,10 @@ export function trapKeyboardNavigation(container: HTMLElement): IDisposable {
 		}
 	});
 }
+
+/**
+ * Convert the SlickGrid's keydown event to VSCode standard keyboard event.
+ */
+export function convertJQueryKeyDownEvent(e: DOMEvent): StandardKeyboardEvent {
+	return new StandardKeyboardEvent((<any>e as JQuery.KeyDownEvent).originalEvent);
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/dom.ts:126]: Why not just have the parameter be a JQuery.KeyDownEvent to begin with? Casting like this is potentially risky...

File: src/sql/base/browser/ui/table/plugins/copyKeybind.plugin.ts [hunk 2]
```diff
@@ -53,7 +53,6 @@ export class CopyKeybind<T> implements Slick.Plugin<T> {
 		if (handled) {
 			e.preventDefault();
 			e.stopPropagation();
-			e.stopImmediatePropagation();
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/table/plugins/copyKeybind.plugin.ts:56]: Why was this removed?

File: src/sql/base/browser/dom.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/base/browser/dom.ts:126]: it is defined like that in the slickgrid's typing, maybe I also modify it.

Comment (alanrenmsft) [reply src/sql/base/browser/dom.ts:126]: it is defined as a generic event: https://github.com/microsoft/azuredatastudio/blob/main/src/typings/slickgrid.d.ts#L65

File: src/sql/base/browser/ui/table/plugins/copyKeybind.plugin.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/base/browser/ui/table/plugins/copyKeybind.plugin.ts:56]: I've reverted the removal, not sure whether they are actually needed, but just in case.

File: src/sql/base/browser/dom.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/base/browser/dom.ts:126]: Bleh, oh well guess it's easy enough to verify we're using it only for keyboard events so guess we'll live with this.

**Review by Charles-Gagnon — APPROVED** [review]
